### PR TITLE
Add sample showing converter delegation via annotations.

### LINKED ISSUE #1444: Question about multi converters
Hi,
I add the converters like this.
```
Retrofit client = new Retrofit.Builder()
           .baseUrl(BASE_URL)
           .client(okClient)
           .addConverterFactory(SimpleXmlConverterFactory.create())
           .addConverterFactory(GsonConverterFactory.create())
           .addCallAdapterFactory(RxJavaCallAdapterFactory.create())
           .build();
```
But, the simplexml will parse the data first and throws a XmlPullParserException when i request a JSON data. The GsonConverter doesn't work.

### PR DESCRIPTION
**Pull request by JakeWharton**
Closes #1444.

### CODE REVIEW

File: samples/src/main/java/com/example/retrofit/JsonAndXmlConverters.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
+package com.example.retrofit;
```
Comment (swankjesse) [inline samples/src/main/java/com/example/retrofit/JsonAndXmlConverters.java]: copyright on samples?

File: samples/src/main/java/com/example/retrofit/JsonAndXmlConverters.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    Call<User> exampleJson();
+    @GET("/") @Xml
+    Call<User> exampleXml();
+  }
+
+  public static void main(String... args) throws IOException {
+    MockWebServer server = new MockWebServer();
+    server.start();
+    server.enqueue(new MockResponse().setBody("{\"name\": \"jason\"}"));
+    server.enqueue(new MockResponse().setBody("<user name=\"Eximel\"/>"));
```
Comment (swankjesse) [inline samples/src/main/java/com/example/retrofit/JsonAndXmlConverters.java]: Cute example

File: samples/src/main/java/com/example/retrofit/JsonAndXmlConverters.java (hunk 1 above)
Comment (JakeWharton) [inline samples/src/main/java/com/example/retrofit/JsonAndXmlConverters.java]: Damn. I was doing so well :(

**Comment by swankjesse** [comment]
LGTM

**Pull request merged**